src/js/player.js

```diff
@@ -823,6 +833,151 @@ class Player extends Component {
+      };
+    }
+
+    // if the tag initially had a source before we started watching
+    // we need to fire sourceset when the player is ready
+    if (this.tagAttributes && this.tagAttributes.src) {
+      this.ready(() => {
+        this.trigger('sourceset');
+      });
+    }
+
+    const setupSrcObserver = () => {
+      const el = this.el().getElementsByTagName('video')[0] || this.el().getElementsByTagName('audio')[0];
+
+      if (!el) {
+        return;
+      }
+
+      Object.defineProperty(el, 'src', {
+        get: srcDescriptor.get.bind(el),
+        set: (v) => {
+          const retval = srcDescriptor.set.call(el, v);
+
+          this.ready(() => this.trigger('sourceset'), true);
+
+          return retval;
+        },
+        configurable: true,
+        enumerable: true
+      });
+
+      el.setAttribute = (n, v) => {
+        const retval = base.setAttribute.call(el, n, v);
+
+        if (n === 'src') {
+          this.ready(() => this.trigger('sourceset'), true);
+        }
+
+        return retval;
+      };
+
+      el.load = () => {
+        const retval = base.load.call(el);
+
+        this.ready(() => this.trigger('sourceset'), true);
+
+        return retval;
+      };
+
+      this.on('dispose', () => {
+        if (el) {
+          el.load = window.HTMLMediaElement.prototype.load.bind(el);
+          el.setAttribute = window.HTMLMediaElement.prototype.setAttribute.bind(el);
+          Object.defineProperty(el, 'src', {
+            get: srcDescriptor.get.bind(el),
+            set: srcDescriptor.set.bind(el),
+            configurable: true,
+            enumerable: true
+          });
+        }
+      });
+    };
+
+    // observer video tag changes, we do this when tech loads on iOS
+    if (window.MutationObserver) {
+      this.videoTagObserver_ = new window.MutationObserver((muts) => {
+        muts.forEach((mut) => {
+          const addedNodes = Array.prototype.filter.call(mut.addedNodes, (node) => (/video|audio/i).test(node.tagName));
+
+          if (!addedNodes.length) {
+            return;
+          }
+          setupSrcObserver();
+        });
+      });
+
+      this.videoTagObserver_.observe(this.el(), {childList: true});
+    }
+    setupSrcObserver();
+    this.on('sourceset', this.handleSourceSet_);
+  }
+
+  /**
+   * update the internal source caches based on the current source that was set
+   */
+  handleSourceSet_(e) {
+    const el = this.el().getElementsByTagName('video')[0] || this.el().getElementsByTagName('audio')[0];
+
+    if (!el) {
+      return;
+    }
+
+    // get what the video element thinks the source is
+    let src = el.src;
+    let sources = Array.prototype.map.call(el.getElementsByTagName('source'), (sourceEl) => {
+      return {src: sourceEl.src, type: sourceEl.type};
+    });
+
+    if (!sources.length && !src) {
+      return;
+    }
+
+    if (el.getAttribute('src') === null) {
+      src = null;
+    }
+
+    if (src) {
+      src = {src, type: 'video/mp4'};
```

Comment:
This is hardcoded for now, we can either use a MIME_TYPE list, or just specify it as unknown. I am also open to suggestions.